Templates/Equality.stencil

```diff
@@ -6,7 +6,7 @@ extension {{ type.name }} {
     func isEqual(to rhs: {{ type.name }}) -> Bool {
         {% for variable in type.storedVariables %}
         {% if variable.typeName.dictionary %}
-        if !NSDictionary(dictionary: {{ variable.name}}{% if variable.typeName.isOptional %} ?? [:]{% endif %}).isEqual(to: rhs.{{ variable.name }}{% if variable.typeName.isOptional %} ?? [:]{% endif %}) { return false }
+        if !NSDictionary(dictionary: {{ variable.name}}{% if variable.typeName.isOptional %} ?? [:]{% endif %}).isEqual(to: NSDictionary(dictionary: rhs.{{ variable.name }}{% if variable.typeName.isOptional %} ?? [:]{% endif %})) { return false }
```

Comment:
I wonder if we have to use `NSDictionary` here 🤔 it's causing issues on Linux

Reply:
I dunno the solution then. Without wrapping in an `NSDictionary`, at least on macOS, these checks can return `false` when it should be `true`.

Reply:
Could you try to drop the `to: ` then? There seems to be function for `NSDictionary`:
```swift
func isEqual(_ object: Any?) -> Bool
```
and
```swift
func isEqual(to otherDictionary: [AnyHashable: Any]) -> [Bool]
```

I think the former could work without compiler errors on both platforms.

Reply:
Done. Let's see if it fixes Linux.

Reply:
Looks like the tests are passing now!

Reply:
great to hear, LGTM 🚀 thanks for the PR 👏